connector.py

```
@@ -0,0 +1,356 @@
+        :param file_path: str: optional path to file. Defaults to None.
+
+        :return [bool]: True if file exists
+        """
+
+        client = self.get_blob_service_client(storage_account=storage_account)
+        blob_client = client.get_blob_client(container, file_path)
+        return blob_client.exists()
+
+    @multi_arguments_decorator(local_support=True)
+    def upload_folder(
+        self,
+        source_path: str = None,
+        source_storage_account: str = None,
+        source_container: str = None,
+        source_file_path: str = None,
+        dest_path: str = None,
+        dest_storage_account: str = None,
+        dest_container: str = None,
+        dest_file_path: str = None,
+    ):
+        """
+        Upload a directory to an azure location. Subdirectories are not currently supported
+
+        :param source_path: str: optional Local path to folder to upload. Defaults to None.
+        :param source_storage_account: str: optional Ignored. Defaults to None.
+        :param source_container: str: optional Ignored. Defaults to None.
+        :param source_file_path: str: optional Ignored. Defaults to None.
+        :param dest_path: str: optional Azure path to upload to. Defaults to None.
+        :param dest_storage_account: str: optional Storage account. Defaults to None.
+        :param dest_container: str: optional Container name. Defaults to None.
+        :param dest_file_path: str: optional Path to folder. Defaults to None.
+
+        :exception ValueError: Raised if source is an Azure path
+        """
+        if self.is_azure_path(source_path):
+            raise ValueError(
+                f"Expected destination to be local path got azure path: {source_path}"
+            )
+
+        container_client = self.get_container_client(
+            storage_account=dest_storage_account, container=dest_container
+        )
+
+        for root, dirs, files in os.walk(source_path):
+            logging.warning(
```

maybe we can throw an exception here if `dirs` is non empty, `subfolders not supported`

---

I thought to instead just do the warning because maybe we still want it to copy stuff int he folder even if it does contain subdirectories, but I am happy either way really.